components/kyma-environment-broker/cmd/subscriptioncleanup/cloudprovider/aws_test.go

```diff
@@ -0,0 +1,26 @@
+package cloudprovider_test
+
+import (
+	"testing"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/cmd/subscriptioncleanup/cloudprovider"
+)
+
+func TestGetEC2Instances(t *testing.T) {
+	// Before running the test, please search for a development aws account credentials
```

Comment:
Unit tests stored together with source code should never target real systems. In this case the unit test could work with a mock client, which responds back with some fake generated data, useful for testing out the cleaner code logic.

- I suggest you to create an interface for the  EC2 Client functions you use in your code, and a Factory interaface which can produce instance of an EC2 Client interface implementation for a given region. 
- Your cleaner struct will be then initialized with an instance of the Factory interface. 
- In the real implementation, you implement the Factory interface like it can be constructed with AWS credentials, and then the `NewClient(region string)` method will initialize and yield a real *ec2.Client, which is compatible with your Client interface.
- In unit test code, you implement a mock factory, whose `NewClient(region string)` yields a mock client implementation, so just generating responses useful for to unit test your code.

On example (from AWS :D) about a simplified version of this technique (without factory, but you will need because you have client per region):
https://aws.amazon.com/blogs/developer/mocking-out-then-aws-sdk-for-go-for-unit-testing/